[v25.1.x] CORE-13747: Add RedpandaRoles to Whole Cluster Recovery

nguyen-andrew opened the pull request:
Backport of PR https://github.com/redpanda-data/redpanda/pull/28166 

Closes #28255

nguyen-andrew committed:
cluster: Adding roles from whole cluster recovery snapshot.

Updating controller_snapshot_reconciler::get_actions to extract
role actions from security snapshot, and then updating the
cluster_recovery_backend to apply the role actions to the
security_frontend. This role recovery will be bundled in with
the ACL recovery stage in order to avoid adding a new recovery
stage enum that would prevent backportablity.

(cherry picked from commit 1746897ee67497d952671c7eef13f910ed2f7e30)

nguyen-andrew committed:
cloud_metadata: Extending WCR unit tests for role recovery.

Extending the existing snapshot reconciliation and cluster recovery
unit tests to exercise role recovery.

(cherry picked from commit 753ef2c45c01b9e26b0afeda040248508a994a18)

nguyen-andrew commented:
/ci-repeat 1

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75260-->
<!--d65294a0-e602-41c9-a077-9904cb48879d-->
<!--3acf049d-34dc-47a5-92ce-42f526ede75c-->
<!--1079aaed-f5e9-42a1-b5ed-482bd15ad814-->
<!--title-->
## Retry command for [Build#75260](https://buildkite.com/redpanda/redpanda/builds/75260)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_invalidate_keys
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--95a97429-c621-4d35-ba79-1a51d78b151d-->
<!--TEST RESULTS-->
<!--d8ae6e44-c5fc-4f9f-b8be-a9eb04dba499-->
<!--TEST RESULTS-->
<!--caf1f698-35ff-4a03-af7d-56be85c871b2-->
<!--TEST RESULTS-->
<!--30f7124d-0681-4de4-b836-c70afa609e0f-->
<!--TEST RESULTS-->
<!--b76f3fc5-45a9-45bd-be34-e0162e4c9cc3-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75260">build#75260</a></summary>


| test_class                          | test_method                        | test_arguments                                                                                                                                                   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                       | test_history                                                                                                                                                                         |
|:------------------------------------|:-----------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest            | test_node_operations               | {"cloud_storage_type": 1, "enable_failures": true, "mixed_versions": false, "with_chunked_compaction": true, "with_iceberg": true, "with_tiered_storage": false} | integration | https://buildkite.com/redpanda/redpanda/builds/75260#019a3271-58f7-463f-9ab6-6430e6b0046c | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS    | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RandomNodeOperationsTest&test_method=test_node_operations                          |
| RedpandaKerberosConfigTest          | test_non_default                   | null                                                                                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/75260#019a326e-7c14-48c4-9120-41aa79ea829c | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RedpandaKerberosConfigTest&test_method=test_non_default                            |
| RedpandaOIDCTlsTest                 | test_admin_invalidate_keys         | null                                                                                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/75260#019a326e-7c13-4dd9-8a0d-f4616eef107f | FAIL          | 0/21     | The test has failed across all retries                                                                                                                       | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RedpandaOIDCTlsTest&test_method=test_admin_invalidate_keys                         |
| RedpandaOIDCTlsTest                 | test_admin_revoke                  | null                                                                                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/75260#019a326e-7c14-48c4-9120-41aa79ea829c | FLAKY         | 8/21     | upstream reliability is '100.0'. current run reliability is '38.095238095238095'. drift is 61.90476 and the allowed drift is set to 50. The test should FAIL | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RedpandaOIDCTlsTest&test_method=test_admin_revoke                                  |
| RedpandaOIDCTlsTest                 | test_admin_whoami                  | null                                                                                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/75260#019a326e-7c15-403c-9b9e-16002cfbd434 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                       | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RedpandaOIDCTlsTest&test_method=test_admin_whoami                                  |
| WriteCachingFailureInjectionE2ETest | test_crash_all_with_consumer_group | null                                                                                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/75260#019a326e-7c19-43be-9384-c8ff9263aaef | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS    | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionE2ETest&test_method=test_crash_all_with_consumer_group |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75336">build#75336</a></summary>


| test_class          | test_method                | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                       | test_history                                                                                                                                                 |
|:--------------------|:---------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RedpandaOIDCTlsTest | test_admin_invalidate_keys | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75336#019a3645-b612-459a-bf91-59d8819d1a80 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                       | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RedpandaOIDCTlsTest&test_method=test_admin_invalidate_keys |
| RedpandaOIDCTlsTest | test_admin_revoke          | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75336#019a3645-b612-459a-bf91-59d8819d1a80 | FLAKY         | 9/21     | upstream reliability is '100.0'. current run reliability is '42.857142857142854'. drift is 57.14286 and the allowed drift is set to 50. The test should FAIL | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RedpandaOIDCTlsTest&test_method=test_admin_revoke          |
| RedpandaOIDCTlsTest | test_admin_whoami          | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75336#019a3645-b612-459a-bf91-59d8819d1a80 | FLAKY         | 1/21     | upstream reliability is '100.0'. current run reliability is '4.761904761904762'. drift is 95.2381 and the allowed drift is set to 50. The test should FAIL   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RedpandaOIDCTlsTest&test_method=test_admin_whoami          |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/76019">build#76019</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                                                   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    | test_history                                                                                                                                                |
|:-------------------------|:---------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 1, "enable_failures": true, "mixed_versions": false, "with_chunked_compaction": false, "with_iceberg": true, "with_tiered_storage": true} | integration | https://buildkite.com/redpanda/redpanda/builds/76019#019a73a7-e3c7-4ca0-b7ea-5f5d055e5120 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RandomNodeOperationsTest&test_method=test_node_operations |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/76083">build#76083</a></summary>


| test_class            | test_method         | test_arguments                                                   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    | test_history                                                                                                                                            |
|:----------------------|:--------------------|:-----------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------|
| PartitionBalancerTest | test_fuzz_admin_ops | null                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/76083#019a7682-986f-4e7f-b678-0097c6dca951 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=PartitionBalancerTest&test_method=test_fuzz_admin_ops |
| SaslPlainTest         | test_plain_authn    | {"client_type": 4, "sasl_plain_enabled": false, "scram_type": 1} | integration | https://buildkite.com/redpanda/redpanda/builds/76083#019a7682-986d-43a1-b27f-788f16f5dc6b | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SaslPlainTest&test_method=test_plain_authn            |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/76142">build#76142</a></summary>


| test_class        | test_method                                | test_arguments                                                     | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                               |
|:------------------|:-------------------------------------------|:-------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DeleteRecordsTest | test_delete_records_concurrent_truncations | {"cloud_storage_enabled": true, "truncate_point": "random_offset"} | integration | https://buildkite.com/redpanda/redpanda/builds/76142#019a7961-4602-452d-bb35-aabd83a50e53 | FLAKY         | 16/21    | upstream reliability is '84.21052631578947'. current run reliability is '76.19047619047619'. drift is 8.02005 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DeleteRecordsTest&test_method=test_delete_records_concurrent_truncations |
</details>
<!--/body-->

nguyen-andrew commented:
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_invalidate_keys
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75336-->
<!--0288cba4-1a28-4356-b3f8-a060d0df90e3-->
<!--title-->
## Retry command for [Build#75336](https://buildkite.com/redpanda/redpanda/builds/75336)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_invalidate_keys
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
```
<!--/body-->

nguyen-andrew commented:
[Force pushed](https://github.com/redpanda-data/redpanda/compare/23c4710048770bdfe6f8da0d32c1750123ac3e94..74c8998e3a951824860d49430beecf1b16633283) to get latest v25.1.x and get the fix from #28451

nguyen-andrew commented:
/ci-repeat 1

nguyen-andrew commented:
[Force pushed](https://github.com/redpanda-data/redpanda/compare/74c8998e3a951824860d49430beecf1b16633283..4c05da6fda4b4d772453d231f983bbef0b0c10db) to add missing functions and classes in rpk.py that haven't been backported yet.

nguyen-andrew committed:
rptest: Backporting rpk.py updates

Backporting one of the updates made to rpk.py in the following commit:
https://github.com/redpanda-data/redpanda/commit/a273f54b22ec8530826f1760437c32a956bd488a

nguyen-andrew committed:
rptest: Backporting rpk.py updates

Backporting updates made to rpk.py in the following commit:
https://github.com/redpanda-data/redpanda/commit/0b0f73a17e311b61e34c9685e29609bd8bea6e2b

nguyen-andrew committed:
rptest: Removing inapplicable fields in backport of RPKACLInput in rpk.py

Removing fields that aren't available to rpk until 25.2.

nguyen-andrew committed:
rptest: Extend controller snapshot recover test for roles

Extending the test_basic_controller_snapshot_restore ducktape test
to validate the restoration of roles.

(cherry picked from commit 7b9eaba2655fb89741b49492d59114860a18e337)

nguyen-andrew commented:
[Force pushed](https://github.com/redpanda-data/redpanda/compare/4c05da6fda4b4d772453d231f983bbef0b0c10db..a38755604e9c0d09143aa99353b407a6f3bc708e) to restructure commits and better separate partial rpk.py backport to help with future archaeology. Also removed inapplicable fields to backport of rpk.py
